### ISSUE TITLE
Fix S1128 FP: Deconstruct method C#

### ISSUE THREAD

**Issue by relrafie** [SonarSource/sonar-dotnet#9747]
### Description

'S1128' Remove this unnecessary 'using'' is raised when ref. is needed for a deconstructed method.

### Reproducer

```
using CSharpFunctionalExtensions;

    private void SomeMethod()
    {
        var (_, _, _) = GetResult();
    }

    private CSharpFunctionalExtensions.Result GetResult()
    {
        return CSharpFunctionalExtensions.Result.Success();
    }
```

Removing the using statement will result in an error, adding it will result in a sonar smell. 

**Expected behavior**
[S1128](https://sonarsource.github.io/rspec/#/rspec/S1128/csharp) shouldn't be detected as there is no error there

**Actual behavior**
A code smell is detected

### Product and Version

SonarQube Server: 2025.1.1.104738; csharpenterprise: "10.4.1 (build 110446)"

**Comment by pavel-mikula-sonarsource** [comment]
Hi @relrafie,

I tried to reproduce this case, but I couldn't get into the scenario with an error, or S1128 raising. Can you try to write both files as a snippets? I tried
```
using CSharpFunctionalExtensions;   // IDE0005 raised
using System.Globalization;         // IDE0005 and S1128 raised

public class Sample
{
    private void SomeMethod()
    {
        var (_, _, _) = GetResult();
    }

    private CSharpFunctionalExtensions.Result GetResult() => null;
}

```
and
```
namespace CSharpFunctionalExtensions;

public class Result
{
    public void Deconstruct(out int a, out int b, out int c)
    {
        a = 42;
        b = 42;
        c = 42;
    }
}
```

**Comment by relrafie** [comment]
Hi @pavel-mikula-sonarsource,
Thank you for attempting to reproduce and I apologize for the incomplete solution. I looked into [CSharpFunctionalExtensions](https://github.com/vkhorikov/CSharpFunctionalExtensions) and it appears that the deconstruct method is actually a static extension method.
So please try the following:

```
using Namespace.Should.Need; // IDE0005 and S1128 raised

public class Sample
{
    private void SomeMethod()
    {
        var (_, _, _) = Namespace.SonarResult.SonarResult.Success();
    }
}

```

```
namespace Namespace.SonarResult;

public readonly struct SonarResult
{
    public SonarResult(bool b, string c)
    {
        B = b;
        C = c;
    }

    public bool A => !B;
    public bool B { get; } = true;
    public string C { get; } = string.Empty;

    public static SonarResult Success()
    {
        return new SonarResult(false, null!);
    }
}
```

where the deconstruct is some extension method for SonarResult here:

```
namespace Namespace.Should.Need;

public static class SonarResultExtensions
{
    public static void Deconstruct(this SonarResult.SonarResult result, out bool a, out bool b,
        out string c)
    {
        a = result.A;
        b = result.B;
        c = result.B ? result.C : default;
    }
}
```

Let me know if you need any more information.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @relrafie,

Thank you for the reproducer. The static extension method was indeed the root cause of the problem that was missing.

I confirm this as a False Positive
